### PR TITLE
Datepicker demo and code examples

### PR DESCRIPTION
**Pull request by Blackbaud-AlexKingman**
#174

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-datepicker-demo](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-datepicker-demo)

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (blackbaud-johnly) [inline src/app/docs/datepicker/datepicker-docs.component.html:18]: On the standard datepicker demo, if I select this checkbox without ever having clicked in the demo, it doesn't take immediate effect. However, if I select the field in the demo before clicking the checkbox (even if I subsequently place focus elsewhere), the field is immediately highlighted with the red error border. The fuzzy datepicker demo on the other hand highlights the field with the red error border when the checkbox is selected regardless of whether the field was ever selected. Is this behavior expected?

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (blackbaud-johnly) [inline src/app/docs/datepicker/datepicker-docs.component.html:10]: Do we really need a separate header and border for each demo? Or could they be combined in the same box with a `Demo` header like how we have the standard checkboxes and radio buttons alongside the icon checkboxes and radio buttons? Not sure if it matters, but just throwing it out there.

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (blackbaud-johnly) [inline src/app/docs/datepicker/datepicker-docs.component.html:103]: Is `Basic` adding any detail here? In the demo, we call it `Standard`. Should we use the same terminology here? And delete `Basic` In the header for the fuzzy datepicker code sample?

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (blackbaud-johnly) [inline src/app/docs/datepicker/datepicker-docs.component.html:33]: Should the datepickers in the demo have labels? I only noticed because I was checking to see if we needed to include an ARIA label in the code sample for accessibility, and I realized that we don't have an ARIA label for this component. Should we create an issue for that? And in the shorter term, would associating the datepickers in the demo help at all from an accessibility-example viewpoint?

**Comment by blackbaud-johnly** [comment]
We have [an issue](https://github.com/blackbaud/skyux-datetime/issues/171) for porting over the Design guidelines for datepicker. Are we good to do that after this PR merges? Or would you like me to set up those Design guidelines in your working branch?

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (Blackbaud-AlexKingman) [reply src/app/docs/datepicker/datepicker-docs.component.html:10]: I'm just going off the design spec: https://blackbaud.invisionapp.com/share/T7RPR60C5NG#/screens/360590459_datepicker

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (Blackbaud-AlexKingman) [reply src/app/docs/datepicker/datepicker-docs.component.html:103]: Yeah that works for me. Updated.

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (Blackbaud-AlexKingman) [reply src/app/docs/datepicker/datepicker-docs.component.html:33]: Good thought. I don't think an ARIA input would apply here because the consumer has control over the actual HTML `input` element. I've updated the demo to show how they would apply a proper label. That should cover us.

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (Blackbaud-AlexKingman) [reply src/app/docs/datepicker/datepicker-docs.component.html:18]: Looks like thats an issue with the datepicker itself. I'll file an issue separately for that.

**Comment by Blackbaud-AlexKingman** [comment]
> We have [an issue](https://github.com/blackbaud/skyux-datetime/issues/171) for porting over the Design guidelines for datepicker. Are we good to do that after this PR merges? Or would you like me to set up those Design guidelines in your working branch?

I'd do that in a separate PR if its all the same for you.

File: src/app/docs/datepicker/datepicker-docs.component.html
Comment (blackbaud-johnly) [reply src/app/docs/datepicker/datepicker-docs.component.html:10]: @Blackbaud-AdamFunderburk, any opinion on this? I assume we're good to push this PR through as is, but just curious if there is a rationale behind when we have multiple demos at the top of the page and when we have multiple items within a single demo. (In this case, I might not even have registered that they were separate except that when I was looking for an ARIA label in the code sample, I realized we don't have an ARIA label for this component and wondered if in lieu of an ARIA label, a regular label would be at least a slight nod toward acccessibility.)

**Review by blackbaud-johnly — APPROVED** [review]
There's an open question for Adam, but I doubt he's going to want to change anything and don't think we need to hold this up since the demo follows the design spec.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-datepicker-demo](https://developer.blackbaud.com/skyux-datetime?_sp.spa=pr-datepicker-demo)

**Pull request merged**
